content/operate/kubernetes/security/vault.md

```diff
@@ -0,0 +1,530 @@
+```
+<VAULT_SECRET_ROOT>/<VAULT_SECRET_PREFIX>/<secret-name>
+```
+
+Default example:
+```
+secret/data/redisenterprise-redis-ns/my-cluster
+secret/data/redisenterprise-redis-ns/my-database-password
+secret/data/redisenterprise-redis-ns/tls-certificates
+```
+
+{{<note>}}
+When using OpenShift, replace `kubectl` commands with `oc` throughout this guide.
+{{</note>}}
+
+## Prerequisites
+
+Before integrating Redis Enterprise operator with HashiCorp Vault, ensure you have the following components properly configured:
+
+HashiCorp Vault Requirements:
+
+- Vault instance: HashiCorp Vault v1.15.2+ with TLS and network connectivity to your Kubernetes cluster
+- Authentication method: Configure Kubernetes authentication method in Vault (see [HashiCorp's Kubernetes Auth documentation](https://developer.hashicorp.com/vault/docs/auth/kubernetes))
+- Secret engine: Enable and configure KV version 2 secret engine
+  - Default mount path: `secret/` (configurable)
+  - Used to store all Redis Enterprise secrets
+  - Supports versioning and metadata
+
+Kubernetes Requirements:
+
+- Vault Agent Injector: Deploy the HashiCorp Vault Agent Injector
+  - Enables automatic secret injection into pods
+  - See [Vault Agent Injector tutorial](https://developer.hashicorp.com/vault/tutorials/kubernetes/kubernetes-sidecar)
+- Network access: Ensure Kubernetes cluster can reach Vault
+  - Configure appropriate network policies and firewall rules
+  - Vault typically runs on port 8200 (HTTPS)
+- Service accounts: Proper RBAC configuration for operator service accounts
+
+Vault editions:
+
+This guide supports both Vault Community and Enterprise editions:
+
+- Vault Community: Use all commands without `-namespace` flags or `VAULT_NAMESPACE` parameters
+- Vault Enterprise: Supports namespaces for logical isolation and multi-tenancy (separate from Kubernetes namespaces)
+
+Minimum token TTL:
+
+Configure Vault token policies with minimum TTL of 1 hour:
+- Prevents frequent token renewal overhead
+- Ensures stable operation during maintenance windows
+- See [Vault token management](https://developer.hashicorp.com/vault/tutorials/tokens/token-management)
+
+### Deployment scenarios
+
+This guide covers the most common deployment scenario with the following assumptions:
+
+- Vault Enterprise with namespace support (adapt for Community Edition by removing namespace parameters)
+- Multiple Redis Enterprise clusters in the same Kubernetes cluster
+- Namespace isolation using Kubernetes namespace suffixes for Vault configurations
+- Production security with proper RBAC and network policies
+
+{{<note>}}
+Multi-cluster considerations: When deploying across multiple Kubernetes clusters with identical namespace names, additional prefixing may be required to avoid Vault path conflicts.
+{{</note>}}
+
+## Configure the operator
+
+1. Configure Vault policies and roles
+
+   Create a policy that grants the Redis Enterprise operator read access to secrets:
+
+   ```bash
```

Comment:
This code block and the one below it don't scroll completely to the right on my laptop screen.
<img width="670" height="696" alt="Screenshot 2025-11-14 at 14 12 29" src="https://github.com/user-attachments/assets/1d9b44c6-d122-4363-a7e7-e58f08591f13" />